apps/vendor-portal/src/modules/offers/pages/offers-details.vue

```diff
@@ -675,17 +672,55 @@ async function onImageUpload(files: FileList) {
   }
 
   files = null;
+};
+
+const onGalleryItemEdit = (item: Image) => {
+  emit("page:open", {
+    component: AssetsDetails,
+    componentOptions: {
+      editableAsset: item,
+      images: offerDetails.images,
+      sortHandler: sortImage,
+    },
+  });
+};
+
+function sortImage(remove = false, localImage: IImage) {
+  const images = offerDetails.images;
+  const image = new Image(localImage);
+  if (images.length) {
+    const imageIndex = images.findIndex((img) => img.id === localImage.id);
+
+    remove ? images.splice(imageIndex, 1) : (images[imageIndex] = image);
+
+    editImages(images);
+  }
 }
 
-function onImageRemove() {
+const editImages = (args: Image[]) => {
+  offerDetails.images = args;
+};
+
+const onGallerySort = (images: Image[]) => {
+  offerDetails.images = images;
+};
+
+const onGalleryImageRemove = (image: Image) => {
   if (
     window.confirm(
       unref(computed(() => t("OFFERS.PAGES.ALERTS.IMAGE_DELETE_CONFIRMATION")))
     )
   ) {
-    offerDetails.imgSrc = undefined;
+    const imageIndex = offerDetails.images.findIndex((img) => {
```

Comment:
I think this logic must be moved out of here into composable level.

Reply:
done